web/src/stream_settings_ui.js

```diff
@@ -470,18 +470,24 @@ export function update_empty_left_panel_message() {
         has_streams =
             stream_data.subscribed_subs().length ||
             $("#streams_overlay_container .stream-row:not(.notdisplayed)").length;
+    } else if (is_not_subscribed_stream_tab_active()) {
+        has_streams =
+            stream_data.unsubscribed_subs().length ||
+            $("#streams_overlay_container .stream-row:not(.notdisplayed)").length;
     } else {
         has_streams = stream_data.get_unsorted_subs().length;
     }
+
     if (has_streams) {
         $(".no-streams-to-show").hide();
         return;
     }
+    $(".no-streams-to-show").children().hide();
     if (is_subscribed_stream_tab_active()) {
-        $(".all_streams_tab_empty_text").hide();
         $(".subscribed_streams_tab_empty_text").show();
+    } else if (is_not_subscribed_stream_tab_active()) {
+        $(".not_subscribed_streams_tab_empty_text").show();
     } else {
-        $(".subscribed_streams_tab_empty_text").hide();
         $(".all_streams_tab_empty_text").show();
```

Comment:
Why removed `.hide()` lines here?

Reply:
"No-streams-to-show" is the parent of all the "empty-text" elements. Therefore, instead of individually hiding each element in every if-else case, it is much better to hide all the elements beforehand and then only show the required element. This approach reduces code duplication as well.

Reply:
Okay, looks good.